src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -168,9 +170,30 @@ export class AddScenarioModal implements OnInit, OnDestroy {
         });
     }
 
+    public equationCheck(check){
+        if (check == true) {
+            this.skipCheck = true;
```

Comment:
Same thing as above, because `check` is a boolean, you can actually set `this.skipCheck = check;` at the beginning of the function.